src/macro_util.rs

```diff
@@ -409,6 +411,31 @@ pub unsafe fn transmute_mut<'dst, 'src: 'dst, Src: 'src, Dst: 'dst>(
     unsafe { &mut *dst }
 }
 
+/// Attempts to transmute `Src` into `Dst`.
+///
+/// A helper for `try_transmute!`.
+#[inline(always)]
+pub fn try_transmute<Src, Dst>(mut src: Src) -> Result<Dst, ValidityError<Src, Dst>>
+where
+    Src: crate::IntoBytes,
+    Dst: crate::TryFromBytes,
+{
+    if !Dst::is_mut_src_valid(&mut src) {
```

Comment:
If we can get `Unalign<Dst>: KnownLayout` without `Dst: KnownLayout`, then we could use `<Unalign<Dst> as TryFromBytes>::try_mut_from` here instead of needing to add `is_mut_src_valid`, right?

Reply:
Using #1252, we might be able to use `Unalign<SizedKnownLayout<Dst>>`.

Reply:
Maybe, though doing so would entail needless runtime checks and ultimately end with throwing away the return value of `try_mut_from`.

I think `is_src_valid` (which I've now generalized to take a `Ptr<Src>`) will also be useful for `try_transmute_mut` and `try_transmute_ref`.